### PR TITLE
the doc for /states has been updated

### PR DESCRIPTION
**Pull request by Arxemond777**
the doc for /states has been updated for performing the search functionality

### CODE REVIEW

File: API.yaml [hunk 1]
```diff
@@ -495,6 +497,10 @@ paths:
                 requested_items:
                 - 1
                 - 2
+                full_search:
+                  true/false
+                regex_map_filters:
+                  ["search_me"]
```
Comment (bhufmann) [inline API.yaml:503]: General comment:
The TSP should be defined in the why it makes sense from the API point of view, independent of the back-end implementation. The front-end and TSP are agnostic to the back-end implementation and definitions in the TSP needs to independent as well. Having said that, we can get inspired by the Trace Compass trace server reference implementation. The names of variables, data structures don't need to match what is done in the Trace Compass back-end.

I'm wondering if the timegraph filter should be an object that has option ;full_search' and the reqex_map_filter in it.

Secondly, there needs to be an description on what is the data provided in the `regex_map_filter` and what is the expected outcome (i.e. back-end reply) when such a filter is applied. Reading the TSP alone, a client or server developer would not know what to pass and what to do with the reply.

**Review by bhufmann — CHANGES_REQUESTED** [review]

File: API.yaml (hunk 1 above)
Comment (bhufmann) [reply API.yaml:503]: I suggest an optional filter data structure `FilterQueryParameters` which has an optional `isFullSearch` and a required `filterExpressionsMap`. 

In the TSP, this data structure `FilterQueryParameters` will be added with key `filter_query_paramters`.

Comment (Arxemond777) [reply API.yaml]: Hi,

Can you see my new PR?

File: API-proposed.yaml [hunk 2]
```diff
@@ -2245,6 +2254,30 @@ components:
...
+      type: object
+      properties:
+        is_full_search:
+          type: boolean
+        filter_expressions_map:
+          type: object
+          properties:
+            key:
+              type: string
+              description: where the key 1 is DIMMED and 4 is EXCLUDED
```
Comment (MatthewKhouzam) [inline API-proposed.yaml:2270]: Either "1"... or can we consider a list of human readable keywords like "Dimmed" and "Excluded"?

File: API.yaml [hunk 3]
```diff
@@ -481,11 +481,15 @@ paths:
       requestBody:
         description: Query parameters to fetch the timegraph states. The object 'requested_timerange'
           is the requested time range and number of samples. The array 'requested_items'
-          is the list of entryId being requested.
+          is the list of entryId being requested. The boolean optional 'is_full_search' if is true
+          then the full time range between the first and last requested timestamp should be searched
+          for filter matches. The map 'filter_expressions_map' where the key 1 is DIMMED and 4 is EXCLUDED,
```
Comment (MatthewKhouzam) [inline API.yaml:486]: same as above

File: API.yaml [hunk 4]
```diff
@@ -1552,6 +1561,30 @@ components:
...
+      type: object
+      properties:
+        is_full_search:
+          type: boolean
+        filter_expressions_map:
+          type: object
+          properties:
+            key:
+              type: string
+              description: where the key 1 is DIMMED and 4 is EXCLUDED
```
Comment (MatthewKhouzam) [inline API.yaml:1577]: Idem

File: API-proposed.yaml [hunk 5]
```diff
@@ -505,6 +509,11 @@ paths:
                 requested_items:
                 - 1
                 - 2
+              filterQueryParameters:
+                is_full_search: true
+                filter_expressions_map:
+                  "1":
+                  - query_for_search
```
[the remark this answers is not in the document]
Comment (Arxemond777) [reply API-proposed.yaml:516]: Hi MatthewKhouzam, 

I did following bhufmann's recommendation above 'Map<Integer, Collection<String>> filterExpressionsMap;'. 
As I understood, the frontend is supposed to rematch based on 'DIMMED = 1 << 0' or 'EXCLUDE = 1 << 2'

File: API.yaml [hunk 6]
```diff
@@ -495,6 +499,11 @@ paths:
                 requested_items:
                 - 1
                 - 2
+              filterQueryParameters:
+                is_full_search: true
+                filter_expressions_map:
+                  "1":
+                  - query_for_search
```
[the remark this answers is not in the document]
Comment (Arxemond777) [reply API.yaml:506]: Hi MatthewKhouzam, 

I did following bhufmann's recommendation above 'Map<Integer, Collection<String>> filterExpressionsMap;'. 
As I understood, the frontend is supposed to rematch based on 'DIMMED = 1 << 0' or 'EXCLUDE = 1 << 2'

File: API-proposed.yaml (hunk 2 above)
Comment (Arxemond777) [reply API-proposed.yaml:2270]: Hi MatthewKhouzam, 

I did following bhufmann's recommendation above 'Map<Integer, Collection<String>> filterExpressionsMap;'. 
As I understood, the frontend is supposed to rematch based on 'DIMMED = 1 << 0' or 'EXCLUDE = 1 << 2'

File: API.yaml (hunk 4 above)
Comment (Arxemond777) [reply API.yaml:1577]: Hi MatthewKhouzam, 

I did following bhufmann's recommendation above 'Map<Integer, Collection<String>> filterExpressionsMap;'. 
As I understood, the frontend is supposed to rematch based on 'DIMMED = 1 << 0' or 'EXCLUDE = 1 << 2'

File: API.yaml (hunk 3 above)
Comment (Arxemond777) [reply API.yaml:486]: Hi MatthewKhouzam, 

I did following bhufmann's recommendation above 'Map<Integer, Collection<String>> filterExpressionsMap;'. 
As I understood, the frontend is supposed to rematch based on 'DIMMED = 1 << 0' or 'EXCLUDE = 1 << 2'

**Comment by MatthewKhouzam** [comment]
Updating Change request, just put clearer examples, "1" is fine.

**Comment by Arxemond777** [comment]
Hi @MatthewKhouzam ,

Please, check my new changes above

**Comment by MatthewKhouzam** [comment]
Also reminder: give an example query like "Duration > 10ms"

**Comment by Arxemond777** [comment]
> Please squash the updates into one patch and put a more descriptive commit message.

done

**Comment by Arxemond777** [comment]
> Also reminder: give an example query like "Duration > 10ms"

it was on the 3rd commit. After squashing it should be visible now

**Comment by bhufmann** [comment]
@hriday-panchasara did you generate the API using swagger UI? If yes, please provide me the link to patch in the Trace Compass incubator to review. If now, please create such patch in the Trace Compass incubator and update this PR accordingly. The procedure is described in the README of this repo.

**Comment by bhufmann** [comment]
Search functionality was covered by #92 specifically commit https://github.com/eclipse-cdt-cloud/trace-server-protocol/pull/92/commits/3471802f0b42c8f98e449f875ac92b0d2f688372. 

Closing this PR. Please reopen if required.

**Pull request closed without merging**